peps/pep-0750.rst

```diff
@@ -1,889 +1,1213 @@
-a motivating example.
+   See `logging.py`__ and `test_logging.py`__.
+   
+   __ https://github.com/davepeck/pep750-examples/blob/main/pep/logging.py
+   __ https://github.com/davepeck/pep750-examples/blob/main/pep/test_logging.py
 
-Function Application
---------------------
 
-Tag strings desugar as follows:
+Example: HTML Templating
+-------------------------
 
-.. code-block:: python
+This PEP contains several short HTML templating examples. It turns out that the
+"hypothetical" ``html()`` function mentioned in the  `Motivation`_ section
+(and a few other places in this PEP) exists and is available in the
+`pep750-examples repository <https://github.com/davepeck/pep750-examples/>`_.
+If you're thinking about parsing a complex grammar with template strings, we 
+hope you'll find it useful.
 
-    mytag'Hi, {name!s:format_spec}!'
 
-This is equivalent to:
+Backwards Compatibility
+=======================
 
-.. code-block:: python
+Like f-strings, use of template strings will be a syntactic backwards incompatibility
+with previous versions.
 
-    mytag(DecodedConcrete(r'Hi, '), InterpolationConcrete(lambda: name, 'name',
-    's', 'format_spec'), DecodedConcrete(r'!'))
 
-.. note::
+Security Implications
+=====================
+
+The security implications of working with template strings, with respect to
+interpolations, are as follows:
 
-    To keep it simple, this and subsequent desugaring omits an important scoping
-    aspect in how names in interpolation expressions are resolved, specifically
-    when defining classes. See `Interpolation Expression Evaluation`_.
+1. Scope lookup is the same as f-strings (lexical scope). This model has been
+   shown to work well in practice.
 
-No Empty Decoded String
------------------------
+2. Code that processes ``Template`` instances can ensure that any interpolations
+   are processed in a safe fashion, including respecting the context in which
+   they appear.
 
-Alternation between decodeds and interpolations is commonly seen, but it depends
-on the tag string. Decoded strings will never have a value that is the empty string:
 
-.. code-block:: python
+How To Teach This
+==================
+
+Template strings have several audiences:
+
+- Developers using template strings and processing functions
+- Authors of template processing code, and
+- Framework authors who build interesting machinery with template strings
+
+We hope that teaching developers will be straightforward. At a glance,
+template strings look just like f-strings. Their syntax is familiar and the
+scoping rules remain the same.
+
+The first thing developers must learn is that template strings don't evaluate
+to strings; instead, they evaluate to a new type, ``Template``. This type is a
+simple type intended to be used by template processing code. It's not until
+developers call *that* code that they get the result they want: typically,
+a string, although processing code can of course return any arbitrary type.
```

Comment:
I like the updated How to Teach this, my only suggestion for an addition would be something along the lines of:

> "Developers need not know the semantics of the ``Template`` type to use t-strings, and can be taught that t-strings must always be used with a consumer function. Similarly to descriptors, many more developers will use t-strings than will write t-string processing functions"

This may be over-egging it a little much, though

Reply:
I'm okay with a little extra egg!

Reply:
I added some egg (and removed a little too). It is hopefully a delicious balance of egg.